x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/execution_plan/required_permissions.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ActionsByType } from './types';
+
+export interface RequiredPermissions {
+  cluster: string[];
+  index: Record<string, string[]>;
+}
+
+export function mergeRequiredPermissions(permissions: RequiredPermissions[]): RequiredPermissions {
+  const clusterSet = new Set<string>();
+  const indexPermissions: Record<string, Set<string>> = {};
+
+  permissions.forEach((permission) => {
+    // Add cluster permissions to the set
+    permission.cluster.forEach((perm) => clusterSet.add(perm));
+
+    // Process index permissions
+    Object.entries(permission.index).forEach(([index, perms]) => {
+      if (!indexPermissions[index]) {
+        indexPermissions[index] = new Set<string>();
+      }
+      perms.forEach((perm) => indexPermissions[index].add(perm));
+    });
+  });
+
+  // Convert sets back to arrays for the return value
+  const result: RequiredPermissions = {
+    cluster: Array.from(clusterSet),
+    index: {},
+  };
+
+  Object.entries(indexPermissions).forEach(([index, permSet]) => {
+    result.index[index] = Array.from(permSet);
+  });
+
+  return result;
+}
+
+export function getRequiredPermissionsForActions(actions: ActionsByType): RequiredPermissions {
+  const permissions: RequiredPermissions[] = [];
+
+  if (actions.upsert_component_template.length > 0) {
+    permissions.push({
+      cluster: ['manage_index_templates'],
+      index: {},
+    });
+  }
+
+  if (actions.delete_component_template.length > 0) {
+    permissions.push({
+      cluster: ['manage_index_templates'],
+      index: {},
+    });
+  }
+
+  if (actions.upsert_index_template.length > 0) {
+    permissions.push({
+      cluster: ['manage_index_templates'],
+      index: {},
+    });
+  }
+
+  if (actions.delete_index_template.length > 0) {
+    permissions.push({
+      cluster: ['manage_index_templates'],
+      index: {},
+    });
+  }
+
+  if (actions.upsert_ingest_pipeline.length > 0) {
+    permissions.push({
+      cluster: ['manage_pipeline'],
```

Comment:
[documentation](https://www.elastic.co/docs/deploy-manage/users-roles/cluster-or-deployment-auth/elasticsearch-privileges) also mentions a `manage_ingest_pipelines` permission that apparently provides the same permissions. wondering if they are aliases and having one or the other works when we explicitly check for `manage_pipeline` privilege ?